File: android/src/main/kotlin/com/techlabs/flutter/flutter_techlabs_library/lib/model/VersionCheckResult.kt
package com.techlabs.flutter.flutter_techlabs_library.lib.model

/**
 * 버전 체크 결과를 나타내는 sealed class.
 *
 * - [ForceUpdate] → callbackCode 1
 * - [OptionalUpdate] → callbackCode 2
 * - [NoUpdateNeeded] → callbackCode 0
 * - [Error] → callbackCode 3
 */
sealed class VersionCheckResult {
    abstract val callbackCode: Int

    data class ForceUpdate(val storePackage: String) : VersionCheckResult() {
        override val callbackCode: Int = 1
    }

    data class OptionalUpdate(val storePackage: String) : VersionCheckResult() {
        override val callbackCode: Int = 2
    }

    data object NoUpdateNeeded : VersionCheckResult() {
        override val callbackCode: Int = 0
    }

    data object Error : VersionCheckResult() {
        override val callbackCode: Int = 3
    }
}

File: lib/src/models/version_check_result.dart
sealed class VersionCheckResult {
  const VersionCheckResult();

  int get callbackCode;
}

final class ForceUpdate extends VersionCheckResult {
  final String storePackage;

  const ForceUpdate(this.storePackage);

  @override
  int get callbackCode => 1;
}

final class OptionalUpdate extends VersionCheckResult {
  final String storePackage;

  const OptionalUpdate(this.storePackage);

  @override
  int get callbackCode => 2;
}

final class NoUpdateNeeded extends VersionCheckResult {
  const NoUpdateNeeded();

  @override
  int get callbackCode => 0;
}

final class VersionCheckError extends VersionCheckResult {
  const VersionCheckError();

  @override
  int get callbackCode => 3;
}
